File: python/semanticabi/metadata/EthTokenType.py
from enum import Enum


class EthTokenType(Enum):

    ETH = ('Eth', False)
    ERC20 = ('Erc20', False)
    ERC721 = ('Erc721', True)
    ERC1155 = ('Erc1155', True)
    CRYPTO_PUNKS = ('CryptoPunks', True)

    code: str
    is_nft: bool

    def __init__(self, code: str, is_nft: bool):
        self.code = code
        self.is_nft = is_nft

File: js/src/semanticabi/metadata/EthTokenType.ts
import {Enum} from "common/Enum";

/**
 * Token types supported in transfers.
 *
 * @author zuyezheng
 */
export class EthTokenType extends Enum {

    static readonly ETH = new this('Eth', false);
    static readonly ERC20 = new this('Erc20', false);
    static readonly ERC721 = new this('Erc721', false);
    static readonly ERC1155 = new this('Erc1155', false);
    static readonly CRYPTO_PUNKS = new this('CryptoPunks', false);

    constructor(
        name: string,
        public readonly isNft: boolean
    ) {
        super(name);
    }

}